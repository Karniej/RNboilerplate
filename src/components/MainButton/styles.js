import { colors } from '../../config'

const styles = {
  textStyle: {
    fontSize: 18,
    color: colors.accentColor
  }
}
export default styles
