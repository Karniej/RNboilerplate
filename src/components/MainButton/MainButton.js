import React from 'react'
import { TouchableOpacity, Text } from 'react-native'
import NavigationActions from 'react-navigation/src/NavigationActions'
import PropTypes from 'prop-types'

import styles from './styles'

const { container, textStyle } = styles

class MainButton extends React.Component {
  navigationHelper = () => {
    const { navigation, routeName, resetStack } = this.props
    if (resetStack) {
      navigation.dispatch(
        NavigationActions.reset({
          index: 0,
          key: null,
          actions: [NavigationActions.navigate({ routeName })]
        })
      )
    }
    navigation.navigate(routeName)
  }

  handleOnPress = () => {
    const { onPress } = this.props
    if (onPress) {
      onPress()
    } else {
      this.navigationHelper()
    }
  }
  render() {
    const { title, additionalStyles } = this.props

    return (
      <TouchableOpacity
        style={[container, additionalStyles]}
        onPress={() => {
          return this.handleOnPress()
        }}
      >
        <Text
          style={textStyle}
        >{title}
        </Text>
      </TouchableOpacity>
    )
  }
}
MainButton.defaultProps = {
  resetStack: false
}

MainButton.propTypes = {
  additionalStyles: PropTypes.object,
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  }),
  resetStack: PropTypes.bool,
  routeName: PropTypes.string,
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func
}

export default MainButton
