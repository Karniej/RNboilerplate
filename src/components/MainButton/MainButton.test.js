import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import MainButton from './MainButton'

configure({ adapter: new Adapter() })

test('Renders correctly, expect onpress to be defined, but not called', () => {
  const defaultProps = {
    navigation: { navigate: jest.fn() },
    onPress: jest.fn(),
    resetStack: false,
    routeName: 'HomeScreen',
    title: 'Open HomeScreen'
  }
  const wrapper = shallow(<MainButton
    {...defaultProps}
  />)
  expect(wrapper).toMatchSnapshot()
})
