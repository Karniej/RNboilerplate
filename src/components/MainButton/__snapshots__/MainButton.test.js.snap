// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly, expect onpress to be defined, but not called 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainButton
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    onPress={[MockFunction]}
    resetStack={false}
    routeName="HomeScreen"
    title="Open HomeScreen"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#1a090d",
            "fontSize": 18,
          }
        }
      >
        Open HomeScreen
      </Text>,
      "onPress": [Function],
      "style": Object {
        "backgroundColor": "#fcc500",
        "borderRadius": 3,
        "padding": 10,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Open HomeScreen",
        "ellipsizeMode": "tail",
        "style": Object {
          "color": "#1a090d",
          "fontSize": 18,
        },
      },
      "ref": null,
      "rendered": "Open HomeScreen",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1a090d",
              "fontSize": 18,
            }
          }
        >
          Open HomeScreen
        </Text>,
        "onPress": [Function],
        "style": Object {
          "backgroundColor": "#fcc500",
          "borderRadius": 3,
          "padding": 10,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Open HomeScreen",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#1a090d",
            "fontSize": 18,
          },
        },
        "ref": null,
        "rendered": "Open HomeScreen",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
